### DISCUSSION TITLE
Traefik Rules

### DISCUSSION THREAD

**Discussion by rquevedo** [Tecnativa/doodba#424] [General]
Hi folks,
When i run new doodba project  instance, traefik is trying to get certificate to rule that starts with 'odoo.projectname.domainname'. For example, for a project named myproject and for domain www.myproject.com show this log:

```bash
,ERRO[2021-05-08T04:01:34Z] Unable to obtain ACME certificate for domains "odoo.myproject.www.myproject.com" detected thanks to rule "Host:odoo.myproject.www.myproject.com"
```
Of course this rule is not present in prod.yaml.
How can i disable this behavior. To many calls to letsencrypt cause rate-limits issues.

Thank in advance.

**Comment by joao-p-marques** [comment]
Hi :wave:

Can you provide your copier answers? Something must be happening there that is forcing that wrong domain...

**Reply by jguenat** [reply]
Hello!
I have the same issue with my prod environment.

Unable to obtain ACME certificate for domains "odoo.doodba13.odoo13.domain.ch" detected thanks to rule "Host:odoo.doodba13.odoo13.domain.ch" ...

I use the inverseproxy from the doc and an unmodified prod.yaml generated by the following copier answers:

```
# Changes here will be overwritten by Copier; NEVER EDIT MANUALLY
_commit: v3.2.0
_src_path: gh:Tecnativa/doodba-copier-template
backup_deletion: false
backup_dst: null
backup_email_from: null
backup_email_to: null
backup_image_version: latest
backup_tz: UTC
cidr_whitelist: null
domains_prod:
- hosts:
  - odoo13.domain.ch
domains_test: null
gitlab_url: null
odoo_dbfilter: .*
odoo_initial_lang: fr_FR
odoo_listdb: true
odoo_oci_image: null
odoo_proxy: traefik
odoo_version: 13.0
paths_without_crawlers:
- /web
- /website/info
postgres_dbname: prod
postgres_username: odoo
postgres_version: 13
project_author: jguenat
project_license: BSL-1.0
project_name: doodba13
smtp_canonical_default: null
smtp_canonical_domains: null
smtp_default_from: null
smtp_relay_host: null
smtp_relay_port: 587
smtp_relay_user: null
smtp_relay_version: '10'
```

Thanks for your insight :)

**Reply by rquevedo** [reply]
Hi @jguenat i solve the issue some time ago and forgot post feedback here. Seems that traefik cache store olds rules, posible from testing scenarios before move to production. I stop traefik container and remove image and volumes(this is important), then start again and problem solved.
Hope this help you.

**Reply by jguenat** [reply]
Thanks for your reply !
I checked again after some time in prod and it seems that those errors where only happening a few times after the initial docker-compose up -d
I don't have them anymore after a few restart.
So not an issue :)

**Reply by andhit-r** [reply]
Hi,

I still encounter this problem on one of my server. Bellow is the screenshot for traefik logs

![Selection_454](https://user-images.githubusercontent.com/1947376/142112326-6e3e778a-7928-4477-857f-35ee91755b5b.png)

Below is the screenshot for my copier answer

![Selection_455](https://user-images.githubusercontent.com/1947376/142112802-2edb8b21-fbcd-437d-83ad-c4082c6fba2a.png)

Below is the prod.yaml file

[prod.yaml.txt](https://github.com/Tecnativa/doodba/files/7551270/prod.yaml.txt)

I already try to docker prune system --all && docker prune system --volumes. Still no luck. This only happens on one server though.

**Reply by joao-p-marques** [reply]
The conflicting domain that is failing when obtaining the certificate doesn't come from the file you provided, so it must be from some other container in your system that is connecting to traefik.

**Reply by andhit-r** [reply]
@joao-p-marques, bellow is the screenshot of docker container ls

![Selection_470](https://user-images.githubusercontent.com/1947376/143153429-30c6321e-ae96-4401-9258-8a7e7db72405.png)

No other container connect to traefik. I recently fire up new AWS EC2 to test this. Same result. Bellow are the relevant files

![Selection_471](https://user-images.githubusercontent.com/1947376/143153886-773a8728-f088-40db-97f0-0d2de7b3c5f7.png)

![Selection_472](https://user-images.githubusercontent.com/1947376/143153896-e66d77ab-2ca9-420e-b203-e5e38ffae40f.png)

[inverseproxy.yaml.txt](https://github.com/Tecnativa/doodba/files/7591963/inverseproxy.yaml.txt)
[copier-answer.yaml.txt](https://github.com/Tecnativa/doodba/files/7591965/copier-answer.yaml.txt)
[prod.yaml.txt](https://github.com/Tecnativa/doodba/files/7591966/prod.yaml.txt)

The first one i use v14 and latest one i use v8.0

**Reply by andhit-r** [reply]
@joao-p-marques additional information from my latest try:

* docker-compose still works. Odoo can be access
* Lets encrypt SSL certificate issued
* Error keeps shows

**Reply by rquevedo** [reply]
I finally remove those errors from my traefik. I have to remove manually(rm -rf /path/to/volume) traefik volumes with traefik down. Then up traefik again and all runs fine. So, the problem is caused by old configs stored in traefik volume, that persist even with a:
`docker-compose down --rmi all -v --remove-orphans`
Hope this help.

**Reply by syci** [reply]
The problem is that the label traefik.frontend.rule is missing: [FQDN] . If we put the tag, it will no longer try to search for the domain according to the default Traefik values.

Overrides the default frontend rule. Default: Host:{containerName}.{domain} or Host:{service}.{project_name}.{domain} if you are using docker-compose.

https://doc.traefik.io/traefik/v1.7/configuration/backends/docker/#labels-overriding-default-behavior
